src/opentslm/model/llm/TimeSeriesLLM.py

```
@@ -36,4 +44,45 @@ def get_eos_token(self) -> str:
+                "Streaming generation currently supports exactly one sample per batch."
+            )
+
+    @staticmethod
+    def _iterate_streamer(streamer: Any, generate_fn: Callable[[], None]) -> Iterator[str]:
+        error: BaseException | None = None
+
+        def runner() -> None:
+            nonlocal error
+            try:
+                generate_fn()
+            except BaseException as exc:  # pragma: no cover - re-raised in caller
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `BaseException`

(BLE001)